tempesta_fw/http.c

```diff
@@ -659,8 +661,12 @@ tfw_http_req_cache_cb(TfwHttpReq *req, TfwHttpResp *resp)
 	spin_unlock(&srv_conn->msg_qlock);
 
 	/* Send request to the server. */
-	if (tfw_connection_send(srv_conn, (TfwMsg *)req, false))
+	if (tfw_connection_send(srv_conn, (TfwMsg *)req, false)) {
+		spin_lock(&srv_conn->msg_qlock);
+		list_del(&req->msg.msg_list);
+		spin_unlock(&srv_conn->msg_qlock);
```

Comment:
Why do we need to delete the message from the list? Server connections are expected to be a contention point to service many concurrent clients, so unnecessary locking is unwished...